pyccel/parser/semantic.py

```diff
@@ -2133,7 +2133,8 @@ def _visit_FunctionalFor(self, expr, **settings):
         indices = expr.indices
         dims    = []
         body    = expr.loops[1]
-
+        #TODO shouldn't be a better way than this ?
+        indices = [sp_Symbol(i.name) for i in indices]
```

Comment:
Is this needed for the atoms statement later?

Reply:
yes the one at line 2237

Reply:
Ok, then I suggest you leave a clear `TODO` referencing issue #694 and I'll fix it when I add the equivalent of `atoms` to `Basic`